bin/s2nc.c

```diff
@@ -571,9 +577,24 @@ int main(int argc, char *const *argv)
 
         /* Save session state from connection if reconnect is enabled */
         if (reconnect > 0) {
-            /* Only need to wait to receive the session ticket in TLS1.3 */
-            if ((conn->actual_protocol_version >= S2N_TLS13 ) && session_ticket) {
+            if (conn->actual_protocol_version >= S2N_TLS13) {
+                if (!session_ticket) {
+                    print_s2n_error("s2nc can only reconnect in TLS1.3 with session tickets.");
+                    exit(1);
+                }
                 GUARD_EXIT(echo(conn, sockfd, &session_ticket_recv), "Error calling echo");
+            } else {
+                if (!session_ticket && s2n_connection_get_session_id_length(conn) <= 0) {
+                    printf("Endpoint sent empty session id so cannot resume session\n");
```

Comment:
This should print to stderr: `fprintf(stderr, "Endpoint sent empty session id so cannot resume session\n");`